chain/jsonrpc-primitives/src/types/chunks.rs

```diff
@@ -0,0 +1,152 @@
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum ChunkReference {
+    BlockShardId(near_primitives::types::BlockId, near_primitives::types::ShardId),
+    BlockIdShardIdStruct(BlockShardId),
+    Hash(near_primitives::hash::CryptoHash),
+    HashStruct(ChunkId),
+}
```

Comment:
Well, this is becoming ugly. Let's have another round of bikeshedding.

To reference a chunk we have two options:

1. Provide a block id (hash or height) and a shard id
2. Provide a chunk hash

There are no other options, so `ChunkReference` should only keep two enum variants. The support of the new API style should be isolated in the `parse` method trying to parse legacy one first, and if that fails, try to parse the new style of params.

Thus, let's go back to two enum variants here:

```suggestion
pub enum ChunkReference {
    BlockIdShardId { block_id: BlockId, shard_id: ShardId },
    ChunkHash { chunk_id: CryptoHash },
}
```